iceoryx_hoofs/include/iceoryx_hoofs/cxx/variant.hpp

```diff
@@ -124,12 +147,23 @@ class variant
 
     /// @brief creates a variant and perform an in place construction of the type T.
     ///         If T is not part of the variant you get a compiler error.
+    /// @tparam[in] T type which should be created inside the variant
+    /// @tparam[in] CTorArguments variadic types of the c'tor arguments
     /// @param[in] type type which should be created inside the variant
     /// @param[in] args variadic list of arguments which will be forwarded to the constructor to
     ///                 the type
     template <typename T, typename... CTorArguments>
     variant(const in_place_type<T>& type, CTorArguments&&... args) noexcept;
 
+    /// @brief creates a variant from a user supplied value
+    /// @tparam[in] T type of the value to be stored in the variant
+    /// @param[in] arg arg to be forwared to the c'tor of T
+    template <typename T,
```

Comment:
Could you try to make this constructor constexpr.

Reply:
Good catch! Made all the c'tors and assignment ops `constexpr` like in the STL.